### PR TITLE
WIP: Implement lazy-load on Identity modules

### PR DESCRIPTION
**Pull request by asg5704**
## Description
Create "code-splitting" and lazy loading on individual components owned by the VSP Identity team.

## Testing done
Manual

## Acceptance criteria
- [x] Loading times are faster and performance is improved
- [x] Identity-related code is abstracted out of the primary `vendor.entry.js` file

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/user/authentication/components/SignInModal.jsx [hunk 1]
```diff
@@ -22,6 +19,17 @@ import environment from 'platform/utilities/environment';
...
+const AlertBox = loadable(() =>
+  import(/* webpackChunkName: "signInModal" */ '@department-of-veterans-affairs/component-library/AlertBox'),
+);
+const Modal = loadable(() =>
+  import(/* webpackChunkName: "signInModal" */ '@department-of-veterans-affairs/component-library/Modal'),
+);
+
+const Telephone = loadable(() =>
+  import(/* webpackChunkName: "signInModal" */ '@department-of-veterans-affairs/component-library/Telephone'),
+);
```
Comment (ncksllvn) [inline src/platform/user/authentication/components/SignInModal.jsx:31]: These design system components are likely being used by all of our application code too, so I feel we don't need to lazy load them?

Comment (AGarcia-Clarity) [reply src/platform/user/authentication/components/SignInModal.jsx]: @ncksllvn I removed any shared modules and instead just lazy loaded non-shared components of Identity.  Sorry this took so long there were a lot of issues with unit-tests & e2e testing in cypress that kept breaking 🤷 .

**Comment by AGarcia-Clarity** [comment]
Request review from @department-of-veterans-affairs/frontend-review-group

**Comment by U-DON** [comment]
Is there a reason this couldn't be done with `React.lazy`? Also seems like the import paths are static in this case, so I think it should have adequate support.

**Comment by AGarcia-Clarity** [comment]
> Is there a reason this couldn't be done with `React.lazy`? Also seems like the import paths are static in this case, so I think it should have adequate support. https://loadable-components.com/docs/loadable-vs-react-lazy/

No reason in particular that it couldn't be done with React.lazy, I've just used the `@loadable/components` before.  I'll convert it to React.lazy re-tag the FE Review Group.

**Review by timwright12 — APPROVED** [review]

File: src/applications/login/containers/SignInApp.jsx [hunk 2]
```diff
@@ -72,7 +82,7 @@ class SignInPage extends React.Component {
     const loggedOut = query.auth === 'logged_out';
 
     return (
-      <>
+      <Suspense fallback={<div>Loading...</div>}>
```
Comment (U-DON) [inline src/applications/login/containers/SignInApp.jsx:85]: Should this use the `LoadingIndicator` from the Design System?

**Review by U-DON — APPROVED** [review]

**Comment by jbalboni** [comment]
@AGarcia-Clarity This looks like a bad merge, you have VAOS changes in the diff here that have already gone into master.

**Comment by AGarcia-Clarity** [comment]
> @AGarcia-Clarity This looks like a bad merge, you have VAOS changes in the diff here that have already gone into master.

@jbalboni Yes, something is going on my local build so I'm creating a new branch/Pull Request and will most likely close this Pull Request.

**Pull request closed without merging**